### PR TITLE
[servicebus] Start development of track 2 for Service Bus

### PR DESCRIPTION
**Pull request by richardpark-msft**
This is basically a verbatim merge (with `--allow-unrelated-histories`) of what's current in the azure-service-bus-go repo. I've moved it into a subfolder where it should reside, but have otherwise made no other source changes. 

For a short period of time we'll be working in both repos, but that should hopefully subside as we prep and release previews out of the monorepo.

(NOTE: I'd previously created a PR for this commit at https://github.com/Azure/azure-sdk-for-go/pull/15216, but this is now retargeting to a separate branch, still within the monorepo).

### CODE REVIEW

File: sdk/servicebus/azservicebus/Makefile [hunk 1]
```diff
@@ -0,0 +1,101 @@
+PACKAGE  = github.com/Azure/azure-service-bus-go
```
Comment (seankane-msft) [inline sdk/servicebus/azservicebus/Makefile:1]: I don't think we need this file anymore

File: sdk/servicebus/azservicebus/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Microsoft Azure Service Bus Client for Golang
+[![Go Report Card](https://goreportcard.com/badge/github.com/Azure/azure-service-bus-go)](https://goreportcard.com/report/github.com/Azure/azure-service-bus-go)
+[![godoc](https://godoc.org/github.com/Azure/azure-service-bus-go?status.svg)](https://godoc.org/github.com/Azure/azure-service-bus-go)
+[![Build Status](https://travis-ci.org/Azure/azure-service-bus-go.svg?branch=master)](https://travis-ci.org/Azure/azure-service-bus-go)
+[![Coverage Status](https://coveralls.io/repos/github/Azure/azure-service-bus-go/badge.svg?branch=master)](https://coveralls.io/github/Azure/azure-service-bus-go?branch=master)
```
Comment (seankane-msft) [inline sdk/servicebus/azservicebus/README.md:5]: These need to be updated once this is merged.

File: sdk/servicebus/azservicebus/changelog.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Change Log
```
Comment (seankane-msft) [inline sdk/servicebus/azservicebus/changelog.md:1]: The changelog should be updated to start from v0.1.0

File: sdk/servicebus/azservicebus/eng/azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
+variables:
```
Comment (seankane-msft) [inline sdk/servicebus/azservicebus/eng/azure-pipelines.yml:1]: I would recommend copy and pasting a ci.yml file from azcore

File: sdk/servicebus/azservicebus/go.mod [hunk 5]
```diff
@@ -0,0 +1,19 @@
+module github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus
```
Comment (seankane-msft) [inline sdk/servicebus/azservicebus/go.mod:1]: Are all of these services required? I don't think autorest is needed in client libraries.

**Review by seankane-msft — APPROVED** [review]
LGTM, left some comments that you can address in later PRs.

File: sdk/servicebus/azservicebus/Makefile (hunk 1 above)
Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/Makefile:1]: I'll delete it - this repo uses a combination of things that aren't "Azure standard" so we'll have to fix those first.

For instance, this Makefile handles things like running terraform, etc... So there's still work that has to happen to get us formally integrated into CI.

File: sdk/servicebus/azservicebus/go.mod (hunk 5 above)
Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/go.mod:1]: I'll have to do a cleanup here. This code doesn't even build yet (that's for the next couple of PRs). This is just a verbatim import.

File: sdk/servicebus/azservicebus/README.md (hunk 2 above)
Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/README.md:5]: It looks like this is the only package in the repo that's like this, which makes me think I should just remove them altogether.

File: sdk/servicebus/azservicebus/eng/azure-pipelines.yml (hunk 4 above)
Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/eng/azure-pipelines.yml:1]: Agreed - I'll need to work that out with @benbp

Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/eng/azure-pipelines.yml:1]: Tracking here: #15245

File: sdk/servicebus/azservicebus/changelog.md (hunk 3 above)
Comment (richardpark-msft) [reply sdk/servicebus/azservicebus/changelog.md:1]: Changelog updated

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**